cnxauthoring/storage/postgresql.py

```diff
@@ -366,3 +366,7 @@ def search(self, limits, type_=Document, submitter_id=None):
             for item in res:
                 yield self._reassemble_model_from_document_entry(**item)
         raise StopIteration
+
+    def restart(self):
+        """Restart the interface"""
+        self.conn = psycopg2.connect(self.conn.dsn)
```

Comment:
Good point.  The simplest thing would be to keep a copy of the original connection string within the class, but will doing this cause a security issue?